Overlay for compat layer EESSI 2023.02

poksumdo opened the pull request:
The overlay for 2022.11 required some changes:
- removed test for `sys-cluster/lmod` in `etc/portage/package.env` because some tests were failing
- new package sets for `eessi-2022.1{0,1}-linux-{aarch64,x86_64}` (package sets for 2022.10 were needed for a previous build for version 2022.10; sets for 2022.10 and 2022.11 are identical; set for 2022.10 might not be needed anymore, was only used for NESSI tests)
  - these package sets are nearly identical to the ones for 2021.12 with the only change that `sys-libs/pam-1.5.2::eessi` has been removed
- archspec ebuild file needed to be updated (adding `DISTUTILS_USE_PEP517=poetry` as described in https://projects.gentoo.org/python/guide/distutils.html#basic-use-pep-517-mode); the Manifest file was updated accordingly
- reframe ebuild file needed to be updated (adding `DISTUTILS_USE_PEP517=setuptools` as described in https://projects.gentoo.org/python/guide/distutils.html#basic-use-pep-517-mode, also removing `dev-python/pyproject2setuppy...`); the Manifest file was updated accordingly

truib committed:
Initial package set for EESSI 2022.10

Just copied from 2021.12

trz42 committed:
Merge pull request #1 from trz42/eessi-2022.10

Initial package set for EESSI 2022.10

truib committed:
Set for EESSI 2022.10 for arch aarch64

truib committed:
Remove sys-libs/pam from new package set

ebuild was removed from overlay. So can't be emerged if it is still
listed in new package sets (for compat layer 2022.10).

truib committed:
Updated to use PEP517 mode

truib committed:
Updated Manifest for updated reframe 3.9.1 ebuild

truib committed:
Updating archspec-0.1.3 ebuild to use PEP517 mode

truib committed:
Changed PEP517 method to poetry

truib committed:
Some tests for lmod fail

Disabling testing lmod for the moment to verify if the installation proceeds.

truib committed:
Sets for eessi-2022.11

boegel commented on etc/portage/package.env:
```diff
@@ -1,3 +1,2 @@
 sys-apps/archspec test.conf
-sys-cluster/lmod test.conf
```
@trz42 Do you have more information on how the Lmod tests were failing?
Can you open an issue with more info, so we can check the details and see if we can re-enable the tests?

poksumdo replied:
Yep. Discussed on Slack Oct 14, 2022. Tests that failed with version 8.6.14
```
diff     R  2.64   rt/avail/avail/t1
diff     R  2.08   rt/changeMPATH/changeMPATH/t1
diff     R  4.19   rt/columnT/columnT/t1
diff     R  0.335  rt/configDir/configDir/t1
diff     R  1.37   rt/csh_swap/csh_swap/t1
diff     R  0.575  rt/deprecation/deprecation/t1
diff     R  1.32   rt/eb/eb/t1
diff     R  0.679  rt/family/family/t1
diff     R  0.247  rt/fish/fish/t1
diff     R  0.302  rt/hook/hook/t1
diff     R  0.963  rt/hook_avail/hook_avail/t1
diff     R  1.11   rt/inherit/inherit/t1
diff     R  0.647  rt/isloaded/isloaded/t1
diff     R  0.575  rt/junk/junk/t1
diff     R  1.35   rt/lars/lars/t1
diff     R  0.383  rt/lastName/lastName/t1
diff     R  8.89   rt/load/load/t1
diff     R  0.741  rt/mpath/mpath/t1
diff     R  0.598  rt/noTCL/noTCL/t1
diff     R  0.407  rt/personal_mf/personal_mf/t1
diff     R  0.621  rt/petsc/petsc/t1
diff     R  0.741  rt/pkg/pkg/t1
diff     R  0.781  rt/purge/purge/t1
diff     R  1.17   rt/reset/reset/t1
diff     R  0.344  rt/rstats/rstats/t1
diff     R  6.74   rt/saverestore/saverestore/t1
diff     R  1.3    rt/sd_tacc/sd_tacc/t1
diff     R  1.83   rt/sh_to_modulefile/sh_to_modulefile/t1
diff     R  0.29   rt/slash/slash/t1
diff     R  3.57   rt/swap/swap/t1
diff     R  2.41   rt/tabcompl/tabcompl/t1
diff     R  1.86   rt/tacc/tacc/t1
diff     R  0.7    rt/uitSitePkg/uitSitePkg/t1
diff     R  0.624  rt/user_inherit/user_inherit/t1

bedroge replied:
Probably similar / related to the issues I experienced as well, see https://github.com/EESSI/compatibility-layer/issues/153.

boegel commented on sys-cluster/reframe/reframe-3.9.1.ebuild:
```diff
@@ -24,11 +33,12 @@ RDEPEND="dev-python/argcomplete[${PYTHON_USEDEP}]
 		>=dev-python/pytest-forked-1.3.0[${PYTHON_USEDEP}]
 	)"
+# removed
+#	dev-python/pyproject2setuppy[${PYTHON_USEDEP}]
```
unless we add a more meaningful comment, just remove these lines?

poksumdo replied:
Removed.

boegel commented on sys-cluster/reframe/reframe-3.9.1.ebuild:
```diff
@@ -2,6 +2,15 @@
+# EAPI=8
+#
+# PYTHON_COMPAT=( python3_{8..10} pypy3 )
+# inherit distutils-r1
```
remove the commented out lines?

poksumdo replied:
Removed.

boegel commented on etc/portage/sets/eessi-2022.11-linux-x86_64:
```diff
@@ -0,0 +1,19 @@
+sys-apps/dbus
+sys-cluster/lmod
+sys-cluster/rdma-core
+=sys-cluster/reframe-3.9.1::eessi
```
@trz42 ReFrame 3.12.0 is the latest version, we should bump to that?

Or, we should remove ReFrame from the compat layer all together, see discussion in https://github.com/EESSI/software-layer/issues/127

poksumdo replied:
If we remove it, can we then still run tests while building the compat layer?

boegel replied:
Sure, we can (temporarily) install `reframe` when we need it (via `pip install`).

poksumdo replied:
Updated to ReFrame 3.12.0

boegel commented on etc/portage/sets/eessi-2022.11-linux-x86_64:
```diff
@@ -0,0 +1,19 @@
+sys-cluster/lmod
+sys-cluster/rdma-core
+=sys-cluster/reframe-3.9.1::eessi
+=sys-fabric/opa-psm2-11.2.205::eessi
```
latest version is 11.2.230, we should update to that?

poksumdo replied:
Added ebuild for 11.2.230.

boegel commented on etc/portage/sets/eessi-2022.11-linux-x86_64:
```diff
@@ -0,0 +1,19 @@
+dev-util/patchelf
+media-fonts/dejavu
+media-fonts/liberation-fonts
+=sys-apps/archspec-0.1.3::eessi
```
Latest version if 0.1.4, we should update to that?

poksumdo replied:
Release notes seem not to indicate that it would add a lot more (https://github.com/archspec/archspec/releases), but sure can be updated.

poksumdo replied:
Updated anyhow.

boegel reviewed (changes requested):
Not sure about the `2022.10` sets, they probably don't belong here since we'll never use them in EESSI?

We should also at least make an attempt to build the compat layer for `ppc64le`, and if that works without trouble deploy it for `2022.11` too (but opt out of building stuff for it in the software layer)

truib committed:
Removed sets for version 2022.10

Those were only temporarily used for tests with NESSI. Likely they will never be used with EESSI. They are also identical to version 2022.11, so could be recreated easily.

truib committed:
Added set for version 2022.11 for CPU family ppc64le

poksumdo commented:
Sets for `2022.10` have been removed. Set `eessi-2022.11-ppc64le` has been added.

boegel commented:
> Sets for `2022.10` have been removed. Set `eessi-2022.11-ppc64le` has been added.

Once we have the companion PR for the [EESSI/compatibility-layer](https://github.com/EESSI/compatibility-layer) repo, I can test the installation on `ppc64le`

truib committed:
Updated archspec to version 0.1.4

truib committed:
Removed unnecessary comments in reframe-3.9.1.ebuild

truib committed:
Adding ebuild for reframe 3.12.0 (currently latest release)

truib committed:
Added opa-psm2-11.2.230.ebuild

truib committed:
Added ebuild file for latest lmod

This is still one release behind what we might want in EESSI. The next
lmod release should support categories which we may make use of in
EESSI. We can use version 8.7.13 for the compat layer to test if it builds.

truib committed:
Modified sets for 2022.11 to use EESSI ebuild for Lmod 8.7.13

truib committed:
Updated required package versions in EESSI sets for 2022.11

truib committed:
Add missing files for Lmod ebuild

Copied from gentoo/gentoo/sys-cluster/lmod/files

truib committed:
Updated Manifest for Lmod ebuild

truib committed:
Added build dependency on sys-devel/bc

truib committed:
Trying if lmod tests still fail for updated version

truib committed:
Add initial patch to fix failing lmod tests

This commit just includes a patch that adds a bit more debug output.
Hopefully the output helps in designing the final patch.

truib committed:
Fixed Manifest

truib committed:
First attempt to fix part of the failing lmod tests

truib committed:
Added sed substitution expressions, removed debug output

Added three more sed substitution expression to catch the remaining
failing tests. Removed temporary debug output.

truib committed:
Fix typo in variable name used in two expressions

trz42 commented:
Patch file `sys-cluster/lmod/files/lmod-8.7.13-fix-failing-tests-with-EESSI.patch` fixes https://github.com/TACC/Lmod/issues/611

truib committed:
Add Lmod-8.7.14 to compat layer

Skip patch for fix failing tests to verify if problem persists in version
8.7.14.

truib committed:
Reintroducing patch for lmod tests for Lmod 8.7.14

truib committed:
Additional packages for EESSI 2022.11

truib committed:
Removed direnv from sets

Seems it is more complicated than thought initially. Can be added later.

bedroge committed:
update set definition to 2023.03

bedroge committed:
remove reframe

bedroge committed:
remove 2022.11 sets

bedroge committed:
ebuild for archspec 0.2.0

bedroge committed:
archspec 0.2.0 checksums

bedroge committed:
ebuild for lmod 8.7.19

bedroge committed:
updates for lmod 8.7.19

bedroge commented:
@trz42 I made a PR https://github.com/trz42/gentoo-overlay/pull/19 that makes some modifications/additions to your PR as a first step towards 2023.02.

trz42 committed:
Merge pull request #19 from bedroge/eessi-2023.02

Updates for EESSI compat layer 2023.02

bedroge commented:
Not sure if we need the different Lmod versions, and the same for ReFrame (as we're kicking them out of the compat layer), but I'll leave them in for now.

boegel reviewed (changes requested):
@bedroge Let's bump `archspec` to 0.2.0 in `scripts/eessi_sets.yml`?

And maybe also specify Lmod version explicitly?

bedroge committed:
bump archspec to 0.2.0

bedroge committed:
make sure to use lmod version 8.7.19 from the eessi overlay

bedroge commented:
> @bedroge Let's bump `archspec` to 0.2.0 in `scripts/eessi_sets.yml`?
> 
> And maybe also specify Lmod version explicitly?

Ah, yes, overlooked that, thanks. Fixed in https://github.com/trz42/gentoo-overlay/pull/20.

trz42 committed:
Merge pull request #20 from bedroge/eessi-2023.02

Few more updates for 2023.02 set
